curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8e4dc6a60f85bf256a0cb.md

```diff
@@ -54,13 +54,13 @@ assert.match(attack.toString(), /Math\.floor\(\s*Math\.random\(\s*\)\s*\*\s*xp\s
 You should add `1` to the result of `Math.floor()`.
 
 ```js
-assert.match(attack.toString(), /Math\.floor\(\s*Math\.random\(\s*\)\s*\*\s*xp(?:\s*\+\s*1)?\s*\)(?:\s*\+ 1)?/);
+assert.match(attack.toString(), /Math\.floor\(\s*Math\.random\(\s*\)\s*\*\s*xp(?:(?:\s*\+\s*1){1}\s*\)|\s*\)(?:\s*\+\s*1){1})/);
 ```
 
 You should add the result of `Math.floor(Math.random() * xp) + 1` to the result of `weapons[currentWeapon].power`.
```

Comment:
```suggestion
You should add the result of `Math.floor(Math.random() * xp) + 1` to the result of `weapons[currentWeaponIndex].power`.
```
Hint update, as in https://github.com/freeCodeCamp/freeCodeCamp/issues/55583#issuecomment-2241605303